### DISCUSSION TITLE
custom pages policy not working

### DISCUSSION THREAD

**Discussion by ibrahim-bedir** [filamentphp/filament#4345] [Help] [answered]
php artisan make:filament-page;

I can't check authorization on the private page, is it my problem?

AuthServiceProvider.php
```php
  protected $policies = [
         'Model' => 'Policy',
        Setting::class => 'App\Policies\SettingPolicy'
    ];

```

**Comment by danharrin** [comment, marked as answer]
Check the policy in the `mount()` method of the page like you would in normal Livewire

**Reply by ibrahim-bedir** [reply]
I can do it this way.
but it still appears in the left menu.

```php 
   public function mount()
      {
          $this->authorize('list.setting', Setting::class);
          $this->loadData();
      }
```

Filament does this dynamically and automatically in Resource.php.
I don't think this happens on custom pages. is it true?

**Reply by danharrin** [reply]
`mount()` only controls the authorisation when the page loads

https://filamentphp.com/docs/2.x/admin/pages/getting-started#conditionally-hiding-pages-in-navigation
